### PR TITLE
system/libuv: Support specify stack size for worker thread

### PR DESCRIPTION
**Pull request by no1wudi**
## Summary
Add a new option to configure stack size of worker thread in thread pool
## Impact
libuv only
## Testing
VELA

### CODE REVIEW

File: system/libuv/0001-libuv-port-for-nuttx.patch [hunk 1]
```diff
@@ -169,7 +169,24 @@ index 869ae95f..8fabeba5 100644
  }
  
  
-@@ -212,7 +217,7 @@ static void init_threads(void) {
+@@ -188,6 +193,16 @@ static void init_threads(void) {
+   const char* val;
+   uv_sem_t sem;
+ 
++  const uv_thread_options_t params = {
++#ifdef DEF_THREADPOOL_STACKSIZE
```
Comment (pkarashchenko) [inline system/libuv/0001-libuv-port-for-nuttx.patch:177]: Seems like `DEF_THREADPOOL_STACKSIZE` is always defined, so why do we need to have `ifdef` here?

Comment (no1wudi) [reply system/libuv/0001-libuv-port-for-nuttx.patch:177]: It's defined on NuttX but not on all other platforms

Comment (pkarashchenko) [reply system/libuv/0001-libuv-port-for-nuttx.patch:177]: yes, but this patch is to build the code with NuttX and not with other platforms :)
Anyway it is minor, so we can leave it as is.

Comment (xiaoxiang781216) [reply system/libuv/0001-libuv-port-for-nuttx.patch:177]: We need ensure the same code still can built on other platform and with the result. And we plan to upstream the change to libuv community, so it's important to make the change as general as possible.

File: system/libuv/0001-libuv-port-for-nuttx.patch [hunk 2]
```diff
@@ -169,7 +169,24 @@ index 869ae95f..8fabeba5 100644
...
++    UV_THREAD_HAS_STACK_SIZE,
++    CONFIG_LIBUV_THREAD_STACKSIZE
```
Comment (pkarashchenko) [inline system/libuv/0001-libuv-port-for-nuttx.patch:179]: @xiaoxiang781216 following the comment above let's apply this change then
```suggestion
+    DEF_THREADPOOL_STACKSIZE
```

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**